passage/src/main/java/id/passage/android/utils/WebViewUtils.kt

```diff
@@ -0,0 +1,61 @@
+package id.passage.android.utils
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Context
+import android.net.Uri
+import android.os.Bundle
+import android.webkit.CookieManager
+import androidx.browser.customtabs.CustomTabsCallback
+import androidx.browser.customtabs.CustomTabsClient
+import androidx.browser.customtabs.CustomTabsIntent
+import androidx.browser.customtabs.CustomTabsServiceConnection
+import androidx.browser.customtabs.CustomTabsSession
+
+object WebViewUtils {
+    private var customTabsSession: CustomTabsSession? = null
+
+    private fun createCustomTabsSession(context: Context): CustomTabsSession? {
+        CustomTabsClient.bindCustomTabsService(
+            context,
+            "com.android.chrome",
+            object : CustomTabsServiceConnection() {
+                override fun onCustomTabsServiceConnected(
+                    name: ComponentName,
+                    client: CustomTabsClient,
+                ) {
+                    customTabsSession =
+                        client
+                            .newSession(
+                                object : CustomTabsCallback() {
+                                    override fun onNavigationEvent(
+                                        navigationEvent: Int,
+                                        extras: Bundle?,
+                                    ) {
+                                        if (navigationEvent == TAB_HIDDEN || navigationEvent == NAVIGATION_FINISHED) {
+                                            CookieManager
+                                                .getInstance()
+                                                .flush()
```

Comment:
Will this clear cookies after every hosted login?

Reply:
The flush method of the CookieManager writes any in-memory cookies to persistent storage, but it does not clear the cookies. This will handle the case if a user close the app and reopen to logout.

Reply:
Amazing, thanks!